### ISSUE TITLE
Pick Utility Type doesn't show error on variable

### ISSUE THREAD

**Issue by GrishaDev** [microsoft/TypeScript#38634]
Hello, looks like there is a bug with Pick Utility Type, giving it incorrect object type should trigger error instantly but nothing happens.
 Happens if you send the object as seperate variable.

```ts
// should pick phone from data object, leaving only phone, accepting only object with phone..
function something(data: Pick<{name: string, phone: number}, "phone">) {
    console.log(data); // prints the data with both name and phone when sent with incorrect type!
}

const idk = {name: "haha", phone: 94835};
something({name: "haha", phone: 94835}); // correct error
something({phone: 94835}); // works as expected
something(idk); // same as first, but no error??

```

**Expected behavior:**
To give error at `something(idk)`

**Actual behavior:**
It eats it like everything is good.

Tried every typescript version on the playground.

**Playground Link:** 
https://www.typescriptlang.org/play?ts=4.0.0-dev.20200517#code/FDD0oAgZwCwewK4BsAmEAOBLAxgaw-AHYCmEAZgE5wC2EKAhgC70RwBGAVsdowDQRJi9AG6ZCAc1aEkATwJwS-etmzF0jMZIWzWnbowgB3TIxjySAOgvAyCQj0wLoNYqc0AKBswBcEAAo4uAA8AN6E9NTEvlCMFJr86ERREIQI1GzEFAC+-ABEiQrEuQB8AJQQIcAQ1RDYClBwghZIcOKeTPSlANwQ4BhxhIxQEKakXixImLikxqYQbHBz4ZEQ9IRoBSQAhMBZIHWEMRCYKPgAvBXLybkw9Le5CUm+AJwALAAcAMwArFldwA1Im4JO4whFrrd7o9Ci8Pj8st1epBMlQKACXMC2iFNsk3l9foi+sQAB7ofTEFDooEwDwnXCEyBQcGrYZkTAUGL8NgIAyEOAQFFwCgAfmFIGqQA

**Comment by nmain** [comment]
Working as intended.  Excess property checks only occur on fresh objects.  It sounds like you want exact types?  See https://github.com/microsoft/TypeScript/issues/12936

**Comment by GrishaDev** [comment]
> Working as intended. Excess property checks only occur on fresh objects. It sounds like you want exact types? See #12936

I disagree, the check should occur on variable objects too, otherwise the Pick is pretty useless..

**Comment by jgbpercy** [comment]
This is definitely working as intended. Your something function is equivalent to the following:
```
function something2(data: {phone: number}) {
    console.log(data); 
}
```
What this means is that you can pass anything to `something` (or `something2`) that has a phone property of type number, but it's allowed to have other properties as well. Your issue is with how function argument checking works, not with Pick.

The line where you have the "correct error" comment is kind of a special case where TS is applying ["excess property checks"](https://www.typescriptlang.org/docs/handbook/interfaces.html#excess-property-checks). In short, when you are creating an object literal of a specific type, then TS will check you don't add extra properties that aren't on that type. The error is not because you can't pass that object to `something`, it's because TS knows that you're creating an object literal of type `{ phone: number }` and have included an "excess property" that's not on that type.

The error you get from this is equivalent:
```
const idk: Pick<{ name: string, phone: number }, "phone"> = {
    name: 'blah',
    phone: 94835,
};
```

Can I ask if the issue you've raised is causing an actual bug in real code? That is possible if you're doing something really involved with `Object.keys` or some such. However, while the above might seem in isolation like an inconsistency/bug in TS, it's actually a fairly carefully considered set of features that make the language meet the _majority_ of people's expectations in the _majority_ of cases.

(also I know this is just example code but please don't store phone "numbers" as numbers 😄 )

**Comment by nitzcard** [comment]
would you reconsider this in 2023? 
We aim for type safety and with Pick things get lost.

You say it's intended behavior, but look at this from this standpoint, 
interface does not accept unknown values without explicitly telling it. 

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4ChnLgB3wBsIAuZAZzClAHMBuLAXyywQHsQq9z0BeTDjyES5AOQAjInDEAaIeyhwQtMsgAc8lq1xCgA
